api/refresh_token_logic.py

```diff
@@ -0,0 +1,48 @@
+from flask import request
+from user_agents import parse
+import requests
+import hashlib
+import time
+import os
+
+def get_user_device():
```

Comment:
Can't we create a class to incapsulate all these stray functions? They seem to serve similar purposes